app/windows/Details/renderer.jsx

```diff
@@ -31,22 +37,78 @@ class DetailsWindow extends React.Component {
 
   constructor(props) {
     super(props)
+
+    this.handleDownload = this.handleDownload.bind(this)
+    this.handleOpenInBrowser = this.handleOpenInBrowser.bind(this)
+    this.handlePin = this.handlePin.bind(this)
+    this.handleUnpin = this.handleUnpin.bind(this)
   }
 
   componentDidMount() {
-    Promise.all([getObjectStat(hash), getObjectDag(hash)])
+    Promise.all([
+      getObjectStat(hash),
+      getObjectDag(hash),
+      isObjectPinned(hash)
+    ])
       .then(result => this.setState({
         stat: result[0],
         dag: result[1],
+        isPinned: result[2]
       }))
   }
 
+  handleDownload() {
+    const options = {
+      title: 'Where should I save?',
+      properties: ['openDirectory'],
+      buttonLabel: 'Save here'
+    }
+
+    // This returns an array
+    const response = remote.dialog.showOpenDialog(remote.app.mainWindow, options)
+    if (response) {
+      const destination = response[0]
+      saveFileToPath(hash, destination)
+    }
+  }
+
+  handleOpenInBrowser() {
+    openInBrowser([hash])
+  }
+
+  handlePin() {
+    pinObject(hash).then(result => this.setState({ isPinned: true }))
+  }
+
+  handleUnpin() {
+    unpinObject(hash).then(result => this.setState({ isPinned: false }))
+  }
+
   render() {
-    const { currentTab, stat, dag } = this.state
+    const { currentTab, isPinned, stat, dag } = this.state
 
     return (
       <Window>
-        <div className='tab-group' style={{minHeight: '27px'}}>
+        <Toolbar title='Properties'>
```

Comment:
In case of a toolbar title, we should probably consider to hide/disable the Window title (and get it as invisible, like in the Storage List, the main window). On macOS it looks weird:

<img width="666" alt="screen shot 2018-03-19 at 19 04 40" src="https://user-images.githubusercontent.com/33528/37613542-917aedca-2ba8-11e8-9e2e-6680e07dd668.png">

What do you think?

Reply:
I will remove the title, since hiding the bar only works on macOS